### DISCUSSION TITLE
The disk space cannot be released after the osd log is deleted

### DISCUSSION THREAD

**Discussion by mc956759** [rook/rook#12289] [Q&A]
I found that my rook-ceph cluster root disk space usage keeping growth.
The command "lsof |grep delete" output :
"ceph-osd  3074146 3075456 osd_srv_h              167   14w      REG              253,0  155319849    3018483 /var/log/ceph/ceph-osd.3.log.1 (deleted)"
After I restart osd-pod, the space is released.
So why are the logs still  hold by osd? How can I release the logs without restarting osd-pod?
Thanks for any help!!

**Comment by peterroth** [comment]
On Linux, you can delete a file, but if a process is holding a file's inode, the inode is kept and the disk space won't be freed up, hence it still consumes the storage.
Probably that's what you see. The file is deleted while the process keeps it open, writes into it. As long as the pod isn't killed, the inode is kept active and the file will be written.
So, either stop the OSD, delete the log file and start the OSD, or delete the file and do a `kubectl - n rook-ceph rollout restart deployment rook-ceph-osd-<ID>`.
